server/src/test/java/io/crate/expression/symbol/SymbolsTest.java

```diff
@@ -66,4 +68,22 @@ public void test_symbols_are_accountable() throws Exception {
         Symbol symbol = e.asSymbol("x + 10 > 100");
         assertThat(symbol.ramBytesUsed()).isEqualTo(920L);
     }
+
+    /*
+     * https://github.com/crate/crate/issues/16887
+     */
+    @Test
+    public void test_cast_text_array_to_object_array() throws IOException {
```

Comment:
imho this could be done in CastFunctionTest as something like:
```
@Test
    public void test_cast_text_array_to_object_array() throws IOException {
        assertEvaluate("tags::ARRAY(OBJECT)",
            List.of(Map.of("x", "foo", "y", 2), Map.of("y", 2, "z", "bar")),
            Literal.of(List.of("{\"x\":\"foo\",\"y\":2}", "{\"y\":2,\"z\":\"bar\"}"), new ArrayType<>(DataTypes.STRING)));

        assertEvaluate("tags::ARRAY(JSON)::ARRAY(OBJECT)",
            List.of(Map.of("x", "foo", "y", 2), Map.of("y", 2, "z", "bar")),
            Literal.of(List.of("{\"x\":\"foo\",\"y\":2}", "{\"y\":2,\"z\":\"bar\"}"), new ArrayType<>(DataTypes.STRING)));
    }
```

Reply:
(it breaks without the fix)

Reply:
In terms of readable I prefer the other version .The test is also not lighter it is still `CrateDummyClusterServiceUnitTest`.

Reply:
The `tags` is already defined in `ScalarTestCase`, no need to add a table, and with this test, you also check the target value, not only the types, but as you like.